### ISSUE TITLE
textDocument/formatting returning wrong offets

### ISSUE THREAD

**Issue by prabirshrestha** [rcjsuen/dockerfile-language-server-nodejs#139]
Here is a sample `Dockerfile`.

```Dockerfile
    FROM openjdk:8-jdk
```

There are 4 spaces in front of FROM. `range.end.character` should be `3` instead of `4`. In LSP the index starts with 0. After running this command it is getting formated as `ROM openjdk:8-jdk` and the first character is missing.

```json
{
  "response": {
    "id": 5,
    "jsonrpc": "2.0",
    "result": [
      {
        "range": {
          "end": {
            "character": 4,
            "line": 0
          },
          "start": {
            "character": 0,
            "line": 0
          }
        },
        "newText": ""
      }
    ]
  },
  "request": {
    "method": "textDocument/formatting",
    "jsonrpc": "2.0",
    "id": 5,
    "params": {
      "options": {
        "insertSpaces": true,
        "tabSize": 4
      },
      "textDocument": {
        "uri": "file:///C:/Users/prshrest/Desktop/Dockerfile"
      }
    }
  }
}
```

**Comment by rcjsuen** [comment]
@prabirshrestha I can't reproduce this in VS Code. According to the [protocol](https://github.com/Microsoft/language-server-protocol/blob/e1ecd6a23a80735a1a7dc2abe3b684a7032e651e/protocol.md)...

> Position in a text document expressed as zero-based line and zero-based character offset. A position is between two characters like an 'insert' cursor in a editor.

`4` is the space between the character at the third index ` ` and the fourth index `F`.

> A range in a text document expressed as (zero-based) start and end positions. A range is comparable to a selection in an editor. Therefore the end position is exclusive.

Since the `end` position is exclusive, I feel that `4` is correct, do you agree?

**Comment by prabirshrestha** [comment]
That makes sense. I fixed `vim-lsp` instead.

What is the extension name that is using this lang server in VsCode. I would like to try it there too so I know it isn't a bug on my code.

**Comment by rcjsuen** [comment]
Thanks for confirming, @prabirshrestha.

There are no published extensions that use this language server at the moment. There is an [open pull request](https://github.com/Microsoft/vscode-docker/pull/100) for vscode-docker but I'm not sure when it will get merged in. You are welcome to try it out that way if you are familiar with how self-hosting works in VS Code.

**Comment by rcjsuen** [comment]
Hi @prabirshrestha, the newly released version 0.0.17 of the vscode-docker extension is using this language server. Please feel free to install it and try things out.
